includes/class-amp-theme-support.php

```diff
@@ -2103,10 +2264,18 @@ public static function prepare_response( $response, $args = [] ) {
 			$dom->documentElement->setAttribute( 'amp', '' );
 		}
 
-		AMP_HTTP::send_server_timing( 'amp_dom_parse', -$dom_parse_start, 'AMP DOM Parse' );
+		// Apply dev mode mutations when in dev mode.
+		if ( $dev_mode_enabled ) {
+			list( $original_admin_bar_element, $placeholder_admin_bar_element ) = self::apply_dev_mode_mutations( $dom, $dev_mode_xpaths );
```

Comment:
It would be somewhat better if this code were placed into a new `AMP_Dev_Mode_Sanitizer`. The challenge here is that the sanitizer needs to process elements and then take some more actions later on (replacing the placeholder with the original admin bar). Nevertheless, since the conversion sanitizers (e.g. `AMP_Img_Sanitizer) are all now configured to skip over elements that have `data-ampdevmode` attributes, then in practice there isn't _really_ a need to temporarily remove the `#wpadminbar` element from the document while sanitization is occuring. We aren't removing other such elements outside of this tree (e.g. scripts and stylesheets). So perhaps better to just go with a new sanitizer which is forcibly set to run at the very beginning (in the same way that the style sanitizer and tag/attribute sanitizer run at the end).

Reply:
Done in d1ac1642707faf4ca28157bf974c7c8aaaf2458f